sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -4136,6 +4136,26 @@ class Dataset[T] private[sql](
     new MergeIntoWriter[T](table, this, condition)
   }
 
+  def update(set: Map[String, Column], where: Column): Unit = {
```

Comment:
We need to add Scaladoc for this method since it is an external method.

Reply:
nit: for method parameter names, shall we use  
```
def update(assignments:Map[String, Column], condition: Column
``` 
to be consistent with the `UpdateTable` logical plan and also the `mergeInto` API?